pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
I haven't been following this PR terribly closely, but I think one
significant disadvantage is that you can't be certain a role change takes
effect across the entire cluster easily (can't manage centrally).  That's
certainly worrying for an administrator.

Multilevel security policy is not uncommon - I would hesitate to say that a
tested multilevel system is no more at risk than a tested flattened system.

Are there other advantages to propagation that inclines you to prefer it?

 If we wanted to make it so that NamespacedRoleBindings could not reference
a ClusterRole, then we could have a controller that watches for new
namespace (like token controller does) and then automatically creates Roles
in each new namespace when the namespace. Those roles would explicitly
include the namespace that they are restricted to, rather than it being
implied, which might somewhat reduce the chance of bugs. This would have
about the same level of convenience. It would also have the property that
the per-namespace default policy would not automatically update when the
cluster is upgraded, which could be good, but could also be bad if it means
that you have to update each namespace's policy when a new Kind is added to
the system that you want to add for certain roles.